lib/_http_server.js

```diff
@@ -512,7 +512,8 @@ function socketOnError(e) {
 
   if (!this.server.emit('clientError', e, this)) {
     if (this.writable) {
-      this.end(badRequestResponse);
+      this.write(badRequestResponse);
+      this.destroy(e);
```

Comment:
Would it make sense to wait for the `write()` callback to be called?

Reply:
@addaleax @mcollina the assumption is that `badRequestResponse` is the first chunk that is written to the socket (it initiates the HTTP response). As such it is written synchronously so I think there is no need to wait for the write callback.